src/Native/WebGL.js

```diff
@@ -426,89 +429,65 @@ var _elm_community$elm_webgl$Native_WebGL = function() {
-      div.style.overflow = 'hidden';
-      var canvas = _evancz$elm_graphics$Native_Element.createNode('canvas');
-      var gl = canvas.getContext('webgl') || canvas.getContext('experimental-webgl');
+  function toHtml(functionCalls, factList, renderables) {
+  	var model = {
+  		functionCalls: functionCalls,
+  		renderables: renderables,
+      cache: {}
+  	};
+  	return _elm_lang$virtual_dom$Native_VirtualDom.custom(factList, model, implementation);
+  }
 
-      if (gl) {
-        A2(List.map, function(functionCall){
-          functionCall(gl);
-        }, functionCalls);
-      } else {
-        div.innerHTML =
-          '<div style="display: table-cell; text-align: center; width: ' + w + 'px; height: ' + h +
-          'px; vertical-align: middle;"><a href="http://get.webgl.org/">Enable WebGL</a> to see this content!</div>';
-      }
+  // WIDGET IMPLEMENTATION
+  var implementation = {
+  	render: renderCanvas,
+  	diff: diff
+  };
 
-      model.cache.gl = gl;
-      model.cache.canvas = canvas;
-      model.cache.shaders = [];
-      model.cache.programs = {};
-      model.cache.buffers = [];
-      model.cache.textures = [];
 
-      update(div, model, model);
+  function renderCanvas(model) {
 
-      return div;
+    LOG("Render canvas");
+	  var canvas = document.createElement('canvas');
+    var gl = canvas.getContext('webgl') || canvas.getContext('experimental-webgl');
 
+    if (gl) {
+      A2(List.map, function(functionCall){
+        functionCall(gl);
+      }, model.functionCalls);
+    } else {
+      canvas = document.createElement('div');
+      canvas.innerHTML = '<a href="http://get.webgl.org/">Enable WebGL</a> to see this content!';
     }
 
-    function update(div, oldModel, newModel) {
-
-      newModel.cache = oldModel.cache;
-
-      var canvas = newModel.cache.canvas;
+    model.cache = model.cache || {};
+    model.cache.gl = gl;
+    model.cache.shaders = [];
+    model.cache.programs = {};
+    model.cache.buffers = [];
+    model.cache.textures = [];
 
-      canvas.style.width = oldModel.w + 'px';
-      canvas.style.height = oldModel.h + 'px';
-      canvas.style.display = "block";
-      canvas.style.position = "absolute";
-      canvas.width = oldModel.w;
-      canvas.height = oldModel.h;
-
-      if (newModel.cache.gl) {
-        drawGL(newModel);
-      } else {
-        div.firstChild.width = newModel.w + 'px';
-        div.firstChild.height = newModel.h + 'px';
-      }
-
-      div.appendChild(canvas);
-
-      return div;
-    }
-
-    var elem = {
-      ctor: 'Custom',
-      type: 'WebGL',
-      render: render,
-      update: update,
-      model: {
-        models: models,
-        cache: {},
-        w: w,
-        h: h
-      }
-    };
+    return canvas;
+  }
 
-    return A3(_evancz$elm_graphics$Native_Element.newElement, w, h, elem);
 
+  function diff(oldModel, newModel) {
+    newModel.model.cache = oldModel.model.cache;
```

Comment:
@eeue56 true, it doesn't diff anything, it just always renders the new model, as it was before. Function is named `diff` because of how it is called in the api.